### PR TITLE
React unit tests

### PR DESCRIPTION
**Pull request by JACornelius**

### CODE REVIEW

**Comment by JACornelius** [comment]
sorry its not ready to review yet. accidentally opened the pr

File: __tests__/TimelineTest.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+			TimelineResultComp, {className: 'successGetTimeline'}, null))).toEqual(true);
+	});
+
+	it("creates fail TimelineResultComp", function() {
+		wrapper.setState({timeline: null, error: 'There was a problem on the server side, please try again later.'});
+		expect(wrapper.containsMatchingElement(React.createElement(
+			TimelineResultComp, {className: 'error'}, null))).toEqual(true);
+
+	});
+
```
Comment (lee-benjamin) [inline __tests__/TimelineTest.js]: newlines

File: __tests__/TimelineResultCompTest.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	})
+
+	it("contains all div components of the timeline table", function() {	
+		wrapper = mount(React.createElement(TimelineResultComp, {timelineResult: renderedTimeline(mockHttpRespText)}, null));
+		expect(wrapper.find('div').length).toEqual(7);
+
+
+		
+	});
+
```
Comment (lee-benjamin) [inline __tests__/TimelineResultCompTest.js]: newline

File: __tests__/TimelineTest.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import {getTimeline} from '../src/js/services';
+import TimelineButton from '../src/js/components';
+import {TimelineResultComp} from '../src/js/components';
+import {renderedTimeline} from '../src/js/services';
+
+
+describe('Timeline', () => {
+	let wrapper;
+	let mockHttpRespText = [{"message":"mackelmorer AND WE DANCEEDDDDDe",
+	"userName":"Josephine Cornelius",
```
Comment (lee-benjamin) [inline __tests__/TimelineTest.js]: formatting

File: __tests__/TimelineTest.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	});
+
+	it("creates TimelineResultComp", function() {
+		expect(wrapper.containsMatchingElement(React.createElement(TimelineResultComp))).toEqual(true);
+	});
+
+	it("wraps the button container, button, and timeline placeholder with a div", function() {
+		expect(wrapper.children.length).toEqual(1);
+		expect(wrapper.find('div').length).toEqual(3);
+
```
Comment (lee-benjamin) [inline __tests__/TimelineTest.js]: in general, remove superfluous new lines

File: __tests__/TimelineTest.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	it("creates success TimelineResultComp", function() {
+
+		wrapper.setState({timeline: mockHttpRespText, error: null});
+		expect(wrapper.containsMatchingElement(React.createElement(
+			TimelineResultComp, {className: 'successGetTimeline'}, null))).toEqual(true);
+	});
+
+	it("creates fail TimelineResultComp", function() {
+		wrapper.setState({timeline: null, error: 'There was a problem on the server side, please try again later.'});
```
Comment (lee-benjamin) [inline __tests__/TimelineTest.js]: is there a reason for having both kinds of quotes?

File: __tests__/TimelineResultCompTest.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+	})
+
+	it("contains image url", function() {
+		wrapper = mount(React.createElement(TimelineResultComp, {timelineResult: renderedTimeline(mockHttpRespText)}, null));
+		expect(wrapper.find('img').length).toEqual(1);
+
+	})
+
+	it("contains link to profile image URL", function() {
+		wrapper = mount(React.createElement(TimelineResultComp, {timelineResult: renderedTimeline(mockHttpRespText)}, null));
```
Comment (lee-benjamin) [inline __tests__/TimelineResultCompTest.js]: this test doesn't test the condition you mentioned. Test for the actual profile image URL. Look over the rest of your tests and make sure your tests give heed to your mocked data.

**Review by lee-benjamin — CHANGES_REQUESTED** [review]
Can you also give me a quick explanation of what `shallow` and `mount` do? After your changes I'll come over we'll talk about the importance of mocking which I forgot to do earlier.

File: __tests__/TimelineResultCompTest.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+import {shallow, mount, render} from 'enzyme';
+import {shallowToJson} from 'enzyme-to-json';
+import {Timeline} from '../src/js/main.js';
+import {getTimeline} from '../src/js/services';
+import {TimelineButton} from '../src/js/components';
+import {TimelineResultComp} from '../src/js/components';
+import {renderedTimeline} from '../src/js/services';
+
+const mockHttpRespText = [{"message":"mackelmorer AND WE DANCEEDDDDDe",
+	"userName":"Josephine Cornelius",
```
Comment (lee-benjamin) [inline __tests__/TimelineResultCompTest.js]: Line up the colons-- more readable

File: __tests__/TimelineTest.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+import {Timeline} from '../src/js/main';
+import {getTimeline} from '../src/js/services';
+import TimelineButton from '../src/js/components';
+import {TimelineResultComp} from '../src/js/components';
+import {renderedTimeline} from '../src/js/services';
+
+describe('Timeline', () => {
+	let wrapper;
+	let mockHttpRespText = [{"message":"mackelmorer AND WE DANCEEDDDDDe",
+		"userName":"Josephine Cornelius",
```
Comment (lee-benjamin) [inline __tests__/TimelineTest.js]: ditto

File: __tests__/TimelineTest.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+	it("creates pending TimelineResultComp", function() {
+		wrapper.setState({timeline: null, error: null});
+		expect(wrapper.containsMatchingElement(React.createElement(
+			TimelineResultComp, {className: "pending"}, null))).toEqual(true);
+	})
+
+	it("creates success TimelineResultComp", function() {
+		wrapper.setState({timeline: mockHttpRespText, error: null});
+		expect(wrapper.containsMatchingElement(React.createElement(
```
Comment (lee-benjamin) [inline __tests__/TimelineTest.js]: this style is funky, either one-line this (though it is a long line) or use a local variable to make it more readable

**Review by lee-benjamin — CHANGES_REQUESTED** [review]

**Comment by JACornelius** [comment]
I cant find your mount, shallow comment nor the mocking comment so I will just put it here.

Since unit test is meant to test one component/method at a time, sometimes methods rely on other services in order to work. In order to keep it a unit test instead of an integration, we will need to mock "outside" objects and services to test our component or method. For example, lots of unit testing that uses twitter, the twitter obj is mocked

For shallow vs mount. Shallow is on layer, none of the children are rendered. Most of my unit tests are mount because of the many children that need to be checked if they are rendered correctly and because I used a lifecycle function, componentDidMount

**Review by lee-benjamin — APPROVED** [review]

**Comment by lee-benjamin** [comment]
my bad about the confusing styling comment, good work.

**Comment by lee-benjamin** [comment]
by the way, get in the habit of doing "squash and merge", pretty sure most teams prefer that.

**Pull request merged**